src/domain/state.ts

```diff
@@ -126,6 +147,16 @@ const stateToLockedSectionsMap: Record<
   },
 };
 
+// SYSTEM role is not allowed to do any modifications to collaborators or documents no matter what state the app is in
+// ADMIN role can add or delete collaborators and documents under certain conditions
+function checkAppIsApprovedAndUserCanAmend(current: Application, identity: Identity): void {
+  const isReviewer = hasReviewScope(identity);
+  const isSystem = hasDacoSystemScope(identity);
+  if ((isReviewer && current.state !== 'APPROVED') || isSystem) {
+    throw new Error('not allowed');
```

Comment:
there's an error handler that's added to the router in the `app.ts` file: https://github.com/icgc-argo/dac-api/blob/develop/src/app.ts#L80

Although I think I can improve on the statuses of some of these errors instead of just using the default 500 in lots of places

Reply:
sounds good! Yes, "not allowed" might be "Forbidden" which is in that error handler already